src/coins.js

```diff
@@ -3,56 +3,151 @@ const typeforce = require('typeforce')
 
 const networks = require('./networks')
 
+/**
+ * @param network
+ * @returns {string} the name of the network. Returns undefined if network is not a value
+ *                   of `networks`
+ */
+function getNetworkName (network) {
+  return Object.keys(networks).find(n => networks[n] === network)
+}
+
+/**
+ * @param network
+ * @returns {Object} the mainnet corresponding to a testnet
+ */
+function getMainnet (network) {
+  switch (network) {
+    case networks.bitcoin:
+    case networks.testnet:
+      return networks.bitcoin
+
+    case networks.bitcoincash:
+    case networks.bitcoincashTestnet:
+      return networks.bitcoincash
+
+    case networks.bitcoingold:
+    // FIXME(https://github.com/BitGo/bitgo-utxo-lib/issues/50): define bitcoingoldTest
+      return networks.bitcoingold
+
+    case networks.bitcoinsv:
+    case networks.bitcoinsvTestnet:
+      return networks.bitcoinsv
+
+    case networks.dash:
+    case networks.dashTest:
+      return networks.dash
+
+    case networks.litecoin:
+    case networks.litecoinTest:
+      return networks.litecoin
+
+    case networks.zcash:
+    case networks.zcashTest:
+      return networks.zcash
+  }
+  throw new TypeError(`invalid network`)
+}
+
+/**
+ * @param network
+ * @returns {boolean} true iff network is a mainnet
+ */
+function isMainnet (network) {
+  return getMainnet(network) === network
+}
+
+/**
+ * @param network
+ * @returns {boolean} true iff network is a testnet
+ */
+function isTestnet (network) {
+  return getMainnet(network) !== network
+}
+
+const networksArray = Object.keys(networks).map(name => networks[name])
+const mainnets = networksArray.filter(isMainnet)
+const testnets = networksArray.filter(isTestnet)
+const mainnetTestnetPairs = new Map(
+  mainnets.map(m => [m, testnets.filter(t => getMainnet(t) === m)])
+)
+
+/**
+ * @param network
+ * @returns {Object|undefined} - The testnet corresponding to a mainnet.
+ *                               Returns undefined if a network has no testnet.
+ */
+function getTestnet (network) {
+  if (isTestnet(network)) {
+    return network
+  }
+  const testnets = mainnetTestnetPairs.get(network)
+  if (testnets === undefined) {
+    throw new Error(`invalid argument`)
+  }
+  if (testnets.length === 0) {
+    return
+  }
+  if (testnets.length === 1) {
+    return testnets[0]
+  }
+  throw new Error(`more than one testnet for ${getNetworkName(network)}`)
```

Comment:
I like it! Very defensive/forward thinking. I assume this is in anticipation of us perhaps adding a `RegNet` or `SigNet` to some coin in the future?

Reply:
Correct.

Once we introduce regnet, we can introduce `getTestnets(Network): []Network`.

Maybe we should actually do that soon :thinking: